### PR TITLE
e2e(upgrades): crossing hello test

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #5393 

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/core/04-channel/upgrades_test.go [hunk 1]
```diff
@@ -235,6 +236,95 @@ func (s *ChannelTestSuite) TestChannelUpgrade_WithFeeMiddleware_Succeeds() {
...
+	// trying to create some inflight packets, although they might get relayed before the upgrade starts
+	t.Run("create inflight transfer packets between chain A and chain B", func(t *testing.T) {
+		transferTxResp := s.Transfer(ctx, chainA, chainAWallet, channelA.PortID, channelA.ChannelID, testvalues.DefaultTransferAmount(chainADenom), chainAAddress, chainBAddress, s.GetTimeoutHeight(ctx, chainB), 0, "")
+		s.AssertTxSuccess(transferTxResp)
+	})
+
+	t.Run("start relayer", func(t *testing.T) {
+		s.StartRelayer(relayer)
+	})
+
```
Comment (colin-axner) [inline e2e/tests/core/04-channel/upgrades_test.go:267]: should this be after initializing upgrades on each end? Wondering if the relayer will pickup the first upgrade attempt and try on the counterparty before the counterparty submits its upgrade attempt

File: e2e/tests/core/04-channel/upgrades_test.go [hunk 2]
```diff
@@ -235,6 +236,95 @@ func (s *ChannelTestSuite) TestChannelUpgrade_WithFeeMiddleware_Succeeds() {
...
+		s.Require().NoError(err)
+		s.Require().Equal(feetypes.Version, version.FeeVersion, "the channel version did not include ics29")
+
+		// extra check
+		feeEnabled, err := s.QueryFeeEnabledChannel(ctx, chainA, channelA.PortID, channelA.ChannelID)
+		s.Require().NoError(err)
+		s.Require().Equal(true, feeEnabled)
+	})
+
+	t.Run("verify channel A upgraded and is fee enabled", func(t *testing.T) {
```
Comment (colin-axner) [inline e2e/tests/core/04-channel/upgrades_test.go:312]:
```suggestion
	t.Run("verify channel B upgraded and is fee enabled", func(t *testing.T) {
```

**Review by colin-axner — APPROVED** [review]

File: e2e/tests/core/04-channel/upgrades_test.go [hunk 3]
```diff
@@ -235,6 +236,95 @@ func (s *ChannelTestSuite) TestChannelUpgrade_WithFeeMiddleware_Succeeds() {
...
+		transferTxResp := s.Transfer(ctx, chainA, chainAWallet, channelA.PortID, channelA.ChannelID, testvalues.DefaultTransferAmount(chainADenom), chainAAddress, chainBAddress, s.GetTimeoutHeight(ctx, chainB), 0, "")
+		s.AssertTxSuccess(transferTxResp)
+	})
+
+	t.Run("start relayer", func(t *testing.T) {
+		s.StartRelayer(relayer)
+	})
+
+	t.Run("execute gov proposals to initiate channel upgrade on chain A and chain B", func(t *testing.T) {
+		var wg sync.WaitGroup
```
Comment (damiannolan) [inline e2e/tests/core/04-channel/upgrades_test.go:269]: Love me a WaitGroup <3

File: e2e/tests/core/04-channel/upgrades_test.go (hunk 1 above)
Comment (damiannolan) [reply e2e/tests/core/04-channel/upgrades_test.go:267]: that makes sense to me

**Review by damiannolan — APPROVED** [review]
Nice one @crodriguezvega 🙏🏻

**Pull request merged**